pkg/processor/spec_processor_test.go

```diff
@@ -11,6 +11,30 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+func TestSpecJsonPathProcessor(t *testing.T) {
+	t.Parallel()
+	sp := SpecProcessor{
+		selfName:       "abc",
+		readyResources: readyResources(),
+		allowedResources: map[smith.ResourceName]struct{}{
+			"res1": {},
+		},
+	}
+	obj := map[string]interface{}{
+		"ref": map[string]interface{}{
+			// TODO: This JsonPath is ignored, fix the regex match expression
+			"slice": "str>$(res1#a.slice[?(@.label==\"label2\")].value)<str",
```

Comment:
@ash2k need to fix the regex in [spec_processor.go#L24](https://github.com/atlassian/smith/blob/master/pkg/processor/spec_processor.go#L24) (or replace regex with something else).